drivers/perf/raspberrypi_axi_monitor.c

```diff
@@ -521,11 +706,18 @@ static int rpi_axiperf_probe(struct platform_device *pdev)
 	struct device *dev = &pdev->dev;
 	struct device_node *np = dev->of_node;
 	struct device_node *fw_node;
+	const struct of_device_id *match;
 
 	state = kzalloc(sizeof(struct rpi_axiperf), GFP_KERNEL);
 	if (!state)
 		return -ENOMEM;
 
+	match = of_device_get_match_data(dev);
+	if (!match)
+		return -ENODEV;
+
+	state->config = match->data;
```

Comment:
`of_device_get_match_data` already gives you the data pointer, so you want to do
```
state->config = of_device_get_match_data(dev);
if (!state->config)
   return -ENODEV;
```
(possibly -EINVAL rather than -ENODEV?)

Reply:
yeah - I noticed that locally when change stopped working